jira-stats-serverless/analyzer/domain/dev_time_calculator.go

```diff
@@ -89,41 +95,56 @@ func calculateWorkingHours(start time.Time, end time.Time) int {
 
 	for currentDay.Year() <= end.Year() && currentDay.YearDay() < end.YearDay() {
 
-		if isWorkingDay(currentDay) {
+		if this.isWorkingDay(currentDay) {
 			totalHours += 8
 		}
 
 		currentDay = currentDay.Add(time.Hour * 24)
 	}
 
-	//if (t.Weekday() != 6 && t.Weekday() != 7) {
-	//	days++
-	//}
 	return totalHours
 }
 
-func adjustDatesToBounds(interval TransitionInterval, start time.Time, end time.Time) TransitionInterval {
+func (this *DaysCalculator) adjustDatesToBounds(interval TransitionInterval, start time.Time, end time.Time) TransitionInterval {
 	if interval.Start.Before(start) {
 		interval.Start = start
 	}
 
-	if interval.End.After(end) {
-		interval.End = end
+	endBound := this.calculateEndBound(end)
+	if endBound.Before(interval.End) {
```

Comment:
As I understood, here you have to try to wider this interval rather than shrink so I guess we should override current `interval.End` only if the `endBound` is later than that.

Reply:
So either bug or misunderstanding from my side.